deepchem/models/tensorgraph/layers.py

```diff
@@ -69,6 +69,26 @@ def shared(self, in_layers):
     """
     raise ValueError("Each Layer must implement shared for itself")
 
+  def __call__(self, *in_layers):
+    if len(in_layers) > 0:
+      layers = []
+      for in_layer in in_layers:
+        if isinstance(in_layer, Layer):
+          layers.append(layer)
+        elif isinstance(in_layer, tf.Tensor):
+          layers.append(TensorWrapper(in_layer))
+        else:
+          raise ValueError("Layer must be invoked on layers or tensors")
+      self.in_layers = layers
+    return self._create_tensor()
+
+
+class TensorWrapper(Layer):
+  """Used to wrap a tensorflow tensor."""
+
+  def __init__(self, out_tensor):
+    self.out_tensor = out_tensor
```

Comment:
Hmm, good point. Will we ever want to use `TensorWrapper` in a `TensorGraph` or only when layers are invoked in pure tensorflow?

I'll go ahead and add in a `create_tensor` since it will be easy though.